### ISSUE TITLE
Getting "Error: XMLHttpRequest is not defined" when doing requests in Node with SDK

### ISSUE THREAD

**Issue by asastre** [directus/directus#8079]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

Version used of @directus/sdk is 9.0.0-rc.92.

Importing the SDK with an import: `import Directus from "@directus/sdk"` is loading the `sdk.esm.js` file and there the `getDefaultAdapter()` function used from axios is not correct because is using the `xhr` adapter both in browser and in node:

```
function getDefaultAdapter() {
  var adapter;
  if (typeof XMLHttpRequest !== 'undefined') {
    // For browsers use XHR adapter
    adapter = xhr;
  } else if (typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]') {
    // For node use HTTP adapter
    adapter = xhr;
  }
  return adapter;
}
```

I've checked axios code and it seems correct in the latest versions (including the version used in the SDK: `0.21.1`)

### To Reproduce

You can see the wrong implementation if you download the published code of `@directus/sdk` `9.0.0-rc.92` clicking the [tarball link](https://registry.npmjs.org/@directus/sdk/-/sdk-9.0.0-rc.92.tgz) provided with `npm view @directus/sdk@9.0.0-rc.92`.

### What version of Directus are you using?

9.0.0-rc.92

### What version of Node.js are you using?

12.18.0

### What database are you using?

indiferent

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu

### How are you deploying Directus?

locally

**Comment by rijkvanzanten** [comment]
@asastre Are you using CommonJS or ESM in Node? I noticed your version of Node is slightly _before_ Node implemented the "exports" flag in package.json.

All of this is very confusing now with Node switching to ESM as well, so appreciate the help debugging 🙂

**Comment by asastre** [comment]
> @asastre Are you using CommonJS or ESM in Node? I noticed your version of Node is slightly _before_ Node implemented the "exports" flag in package.json.
> 
> All of this is very confusing now with Node switching to ESM as well, so appreciate the help debugging

@rijkvanzanten we're transpiling our code to CommonJS but since we're using the `import` syntax to use Directus (not `require`), the entrypoint used is `sdk.esm.js` which contains the bug explained in the description.

For what I understand, it shouldn't matter if I'm using CommonJS or ESM in Node since the bug is present in the ESM entry point published in this version of Directus, so everyone importing Directus through an `import` statement will get the same error, am I right?

**Comment by rijkvanzanten** [comment]
> For what I understand, it shouldn't matter if I'm using CommonJS or ESM in Node since the bug is present in the ESM entry point published in this version of Directus

You wouldn't believe how finicky all this got now with different bundlers using different entry points and slightly different specs, typescript vs non-tsc, etc etc. Gosh @nickrum and I have been discussing solutions for this for hours 😭 

I think the best approach is going to be to have dedicated `@directus/sdk/browser` and `@directus/sdk/node` exports that are pre-built for the environment, and exclusively build to ESM 🤔

**Comment by joselcvarela** [comment]
I also had stumble on this with Nuxt 3 and @asastre is right, the `sdk.esm.js` is already wrong.
[The solution](https://stackoverflow.com/a/68075268) seems to require `axios` as CJS  and not as `ESM`.
@nickrum do you have any suggestion?

**Comment by joselcvarela** [comment]
Investigating a little further I found that we are building `iife`, `umd`, `esm` and `system` for browser environment with the help of rollup.
Those generated bundles will always have `adapter = xhr;` because the `browser` config on rollup is set to true.

I also found that we are building `cjs` with the help of tsc. In this case, `axios` is not bundled. Instead, it is on the `node_modules`, so the `getDefaultAdapter` should not be an issue in this case.

So the solution should be fix the module resolution which is already done thanks to #9502 later fixed by #9535 by @nickrum .

___
> If you think this is not solved yet, feel free to reach us in order to reopen.
